eng/pipelines/dailytests.yml

```
@@ -0,0 +1,31 @@
+resources:
+  pipelines:
+  - pipeline: ComponentBuildUnderTest
+    source: NuGet.Client-Official
```

> Do we want a PrivateDev version for the daily apex tests or do we just want to keep an official one?

We need a way to run the pipeline for PRs that add new tests, or attempt to fix failing tests, can be validated before merging. I think it would be terrible to need to run an official build for the PR branch, just to test the changes.

However, I don't think we need to run the scheduled tests on both official builds and PR builds. Honestly, I think we should use the PR/private builds as the source/trigger, not official builds. We're running those twice a day already, to calculate build reliability.

```suggestion
    source: NuGet.Client-PrivateDev
```

---

Just to confirm. 

- DailyTests will run based off of the PrivateDev definition as that's really the PR validation, where we're most likely to care about. 
- DailyTests will *not* run as part of the official definition.
- For "validation" purposes, we can use the DailyTests run *against* the dev.

---

> DailyTests will run based off of the PrivateDev definition as that's really the PR validation, where we're most likely to care about.

Yes, I think so (I'm not 100% sure I'm reading your sentence the way you intended, but I think I am). But we also already have the dev branch scheduled twice a day in order to measure build reliability. So, the PrivateDev build definition is not just for PR builds, but also these twice a day scheduled builds.

> DailyTests will not run as part of the official definition.

Right. The daily tests are testing the product, not a specific build definition. PrivateDev and Official both build the same product code. So, in my mind, it doesn't matter which build definition we trigger from.

> For "validation" purposes, we can use the DailyTests run against the dev.

I have low confidence I understand what you mean by this. I assume by validation, you mean a pull request that modifies a daily test that is not run as part of the apex tests that run in the PrivateDev and Official pipelines. Also, if by "run against the dev" you mean "run against the PrivateDev PR build", then yes.

> Just to confirm what you're proposing

If my suggestion isn't accepted, then while I understand the intent of the new pipeline testing official builds, my question is how do we validate pull requests that modify daily tests, given the PrivateDev build definition is not going to run the daily tests?

As mentioned earlier, I'm exploiting the fact that we run dev branch builds in the PrivateDev pipeline regularly, so we can use it as a way to both automatically test the latest dev branch commits, as well as manually triggering the pipeline to test pull requests that modify daily tests.

---

> I have low confidence I understand what you mean by this. I assume by validation, you mean a pull request that modifies a daily test that is not run as part of the apex tests that run in the PrivateDev and Official pipelines. Also, if by "run against the dev" you mean "run against the PrivateDev PR build", then yes.

@zivkan I was more thinking, are we going to be able to track pass rates and avoid these extend packages to be out of sight, out of mind.

In the end, the idea with daily tests is that they should be passing on dev too, as a reference.

---

Both the official build and PrivateDev pipelines build the same source code. So, the dev branch in one is the same as the dev branch in the other.

> these extend packages

I assume this is a typo, and you mean other branches?

If you want to find the test pass rate for the dev branch, ignoring all pull request branches and builds, then if you're using Azure Pipeline Analytics, use the filter and only select the dev branch.

---

I meant to say extended tests. 
I'm aware of the tools, I just wanted to make sure we're gonna get runs to validate that and we're actually gonna pay attention to them :D 
Anyway, I think we're going in circles here, I'm happy with it as is. 

We should *definitely* extend the documentation around this in the client engineering repo though. Describing basically what we discussed here.

---

oh, you mean email notifications when a pipeline fails?   that's something we have to manually configure. If we had different pipelines for automatic daily runs and a different pipeline for validate a PR, then we'd set up an alert for all failires on the daily pipeline.  If we we a single pipeline for both scheduled runs, and also manually triggered PR branches, then I think we'd just need to configure where branch == dev in the email notification settings.